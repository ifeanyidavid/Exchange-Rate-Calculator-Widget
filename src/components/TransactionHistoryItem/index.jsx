import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { ReactComponent as RepeatIcon } from "../../assets/rep.svg";

const Flex = styled.div`
  display: flex;
  justify-content: ${props =>
    props.justifyContent ? props.justifyContent : "flex-start"};

  .exchange-icon {
    display: block;

    flex-shrink: 0;
    border-radius: 50%;
    width: 32px;
    height: 32px;
    display: flex;
  }

  .exchange-icon svg {
    max-width: 100%;
    max-height: 100%;
    width: 100%;
    height: 100%;
  }

  .exchange-icon-dark {
    background: rgba(0, 0, 0, 0.57);
  }
`;

function TransactionHistoryItem({ history, marginBottom }) {
  return (
    <div className={`${marginBottom}`}>
      <Flex justifyContent="space-between">
        <Flex>
          <i className="exchange-icon exchange-icon-dark">
            <RepeatIcon />
          </i>
          <div className="text-left ml-3">
            <span className="d-block">{history.description}</span>
            <span className="d-block text-muted">{history.time}</span>
          </div>
        </Flex>
        <div className="text-right">
          <span className="d-block">
            -{history.sourceCurrency} {history.sourceAmount}
          </span>
          <span className="d-block text-primary">
            +{history.destinationCurrency} {history.destinationAmount}
          </span>
        </div>
      </Flex>
    </div>
  );
}

export default TransactionHistoryItem;

TransactionHistoryItem.propTypes = {
  history: PropTypes.object,
  marginBottom: PropTypes.string
};
