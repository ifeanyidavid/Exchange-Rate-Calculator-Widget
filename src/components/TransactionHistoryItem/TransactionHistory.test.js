import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import TransactionHistoryItem from ".";

configure({ adapter: new Adapter() });

describe("TransactionHistoryItem component", () => {
  it("Renders correctly", () => {
    const history = {
      id: 0,
      description: `Exchanged to EUR`,
      time: new Date().toLocaleTimeString("en-GB"),
      sourceAmount: 23,
      destinationAmount: 42,
      sourceCurrency: "USD",
      destinationCurrency: "EUR"
    };
    const props = {
      history,
      marginBottom: "mb-2"
    };
    // wrapper instance around rendered output
    const wrapper = shallow(<TransactionHistoryItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
