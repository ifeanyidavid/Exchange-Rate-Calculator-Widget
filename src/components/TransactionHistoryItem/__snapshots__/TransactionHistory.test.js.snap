// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionHistoryItem component Renders correctly 1`] = `
<div
  className="mb-2"
>
  <styled.div
    justifyContent="space-between"
  >
    <styled.div>
      <i
        className="exchange-icon exchange-icon-dark"
      >
        <ForwardRef(SvgRep) />
      </i>
      <div
        className="text-left ml-3"
      >
        <span
          className="d-block"
        >
          Exchanged to EUR
        </span>
        <span
          className="d-block text-muted"
        >
          2:35:40 PM
        </span>
      </div>
    </styled.div>
    <div
      className="text-right"
    >
      <span
        className="d-block"
      >
        -
        USD
         
        23
      </span>
      <span
        className="d-block text-primary"
      >
        +
        EUR
         
        42
      </span>
    </div>
  </styled.div>
</div>
`;
